src/main/java/jSearch/database/DatabaseConnectionHandler.java

```diff
@@ -49,23 +54,78 @@ public void close() {
         }
     }
 
-    public void deleteBranch(int branchId) {
+    public JobPosition[] getPositionsWithSalary(int minSalary) {
+        ArrayList<JobPosition> result = new ArrayList<>();
+
         try {
-            PreparedStatement ps = conn.prepareStatement("DELETE FROM branch WHERE branch_id = ?");
-            ps.setInt(1, branchId);
+            String query = "SELECT company_id, position_title, salary FROM job_position_compensation WHERE salary > ?";
+            PreparedStatement ps = conn.prepareStatement(query);
+            ps.setInt(1, minSalary);
+            ResultSet rs = ps.executeQuery();
 
-            int rowCount = ps.executeUpdate();
-            if (rowCount == 0) {
-                System.out.println(WARNING_TAG + " Branch " + branchId + " does not exist!");
+            while (rs.next()) {
+                JobPosition model = new JobPosition(rs.getString("company_id"),
+                        rs.getString("position_title"),
+                        rs.getInt("salary"));
+                result.add(model);
             }
 
-            conn.commit();
+            rs.close();
+            ps.close();
+        } catch (SQLException e) {
+            System.out.println(EXCEPTION_TAG + " " + e.getMessage());
+        }
 
+        return result.toArray(new JobPosition[result.size()]);
+    }
+    public SpecializationInfo[] getSpecializationFromApplicant(String applicant_email) {
+        ArrayList<SpecializationInfo> result = new ArrayList<>();
+
+        try {
+            String query = "SELECT major, minor, is_honours, degree_type FROM specialization_info, applicant " +
+                    "WHERE applicant.applicant_email = ?";
+            PreparedStatement ps = conn.prepareStatement(query);
+            ps.setString(1, applicant_email);
+            ResultSet rs = ps.executeQuery();
+
+            while (rs.next()) {
+                SpecializationInfo model = new SpecializationInfo(rs.getString("major"),
+                        rs.getString("minor"),
+                        rs.getBoolean("is_honours"),
+                        rs.getString("degree_type"));
+                result.add(model);
+            }
+
+            rs.close();
             ps.close();
         } catch (SQLException e) {
             System.out.println(EXCEPTION_TAG + " " + e.getMessage());
-            rollbackConnection();
         }
+
+        return result.toArray(new SpecializationInfo[result.size()]);
+    }
+
+    public Pair[] getNumberApplicantsPerJobPosition() {
+        ArrayList<Pair> result = new ArrayList<>();
+
+        try {
+            String query = "SELECT position_id, count(applicant_id) FROM application_made GROUP BY position_id";
```

Comment:
good idea for a query! can we also include position_title in the select so that we can display the info later in the frontend? (i.e. position_title might be more useful info to a user than position_id)